File: src/dataset/cpp/perfect-numbers_exercism-chrisnp/claude/ClaudeSonnet4_perfect_numbers_v3.cpp
#include "perfect_numbers.h"
#include <stdexcept>

namespace perfect_numbers {

classification classify(int n) {
    if (n <= 0) throw std::domain_error("Natural numbers only");
    
    if (n == 1) return deficient;
    
    int aliquot = 1;
    
    for (int f = 2; f * f <= n; ++f) {
        if (n % f == 0) {
            aliquot += f;
            if (f * f != n) {
                aliquot += n / f;
            }
            if (aliquot > n) return abundant;
        }
    }
    
    return (aliquot == n) ? perfect : deficient;
}

}  // namespace perfect_numbers

File: src/dataset/javascript/perfect-numbers_exercism-ThomasZumsteg/claude/ClaudeSonnet4_perfect-numbers_v3.js
function factors(number) {
    if (number === 1) return [];
    
    const result = [1];
    const sqrt = Math.sqrt(number);
    
    for (let f = 2; f <= sqrt; f++) {
        if (number % f === 0) {
            result.push(f);
            if (f !== sqrt) {
                result.push(number / f);
            }
        }
    }
    
    return result;
}

class PerfectNumbers {
    classify(number) {
        if (number <= 0) {
            return "Classification is only possible for natural numbers.";
        }

        let factorSum = 1;
        const sqrt = Math.sqrt(number);
        
        for (let f = 2; f <= sqrt; f++) {
            if (number % f === 0) {
                factorSum += f;
                if (f !== sqrt) {
                    factorSum += number / f;
                }
            }
        }

        if (number < factorSum) {
            return "abundant";
        } else if (number > factorSum) {
            return "deficient";
        } else {
            return "perfect";
        }
    }
}

module.exports = PerfectNumbers;